spotify/models/user.py

```diff
@@ -122,23 +203,27 @@ def _raise(*args, **kwargs):
 
         return [klass(self.__client, item) for item in resp["items"]]
 
-    async def _refreshing_token(self, expires: int, token: str):
+    async def _refreshing_token(self, expires: int, token: str) -> None:
         while True:
             await asyncio.sleep(expires - 1)
 
-            client_id = self.client.http.client_id
-            client_secret = self.client.http.client_secret
+            self.token_info = await _refresh_token(self.client, token)
 
-            headers = {
-                "Authorization": f"Basic {b64encode(':'.join((client_id, client_secret)).encode()).decode()}",
-                "Content-Type": "application/x-www-form-urlencoded",
-            }
+            self._save_to_cache(self.token_info)
+
+            expires = self.token_info.expires_in
+            self.http.token = self.token_info.access_token
 
-            route = ("POST", REFRESH_TOKEN_URL.format(refresh_token=token))
-            data = await self.client.http.request(route, headers=headers)
+    def _save_to_cache(self, token_info: TokenInfo) -> bool:
```

Comment:
This looks like it belongs as a method of TokenInfo, I suggest moving all caching logic into there.